beacon-chain/core/altair/epoch_precompute.go

```diff
@@ -156,13 +164,13 @@ func ProcessEpochParticipation(
 		return nil, nil, err
 	}
 	for i, b := range pp {
-		if HasValidatorFlag(b, sourceIdx) {
+		if HasValidatorFlag(b, sourceIdx) && vals[i].IsActiveCurrentEpoch {
 			vals[i].IsPrevEpochAttester = true
```

Comment:
This is a bit confusing to me. If the validator is active in the current epoch, there is a chance they weren't current in the previous epoch, so how could they be a previous epoch attester with absolute certainity?

Reply:
Nice catch! the current and previous got switched